docs/standard/serialization/snippets/system-text-json-how-to/csharp/RoundtripDataTable.cs

```diff
@@ -0,0 +1,246 @@
+                            if (DateTime.TryParse(valueString, out DateTime timeSpanValue))
+                            {
+                                jsonWriter.WriteString(key, timeSpanValue.ToString());
+                            }
+                            else
+                            {
+                                jsonWriter.WriteString(key, "");
+                            }
+                            break;
+                        case "System.Byte":
+                        case "System.SByte":
+                        case "System.Decimal":
+                        case "System.Double":
+                        case "System.Single":
+                        case "System.Int16":
+                        case "System.Int32":
+                        case "System.Int64":
+                        case "System.UInt16":
+                        case "System.UInt32":
+                        case "System.UInt64":
+                            if (long.TryParse(valueString, out long intValue))
+                            {
+                                jsonWriter.WriteNumber(key, intValue);
+                            }
+                            else
+                            {
+                                double.TryParse(valueString, out double doubleValue);
+                                jsonWriter.WriteNumber(key, doubleValue);
+                            }
+                            break;
+                        default:
+                            jsonWriter.WriteString(key, valueString);
+                            break;
+                    }
+                }
+                jsonWriter.WriteEndObject();
+            }
+            jsonWriter.WriteEndArray();
+        }
+    }
+    public static class Extensions
+    {
+        public static DataTable JsonElementToDataTable(this JsonElement dataRoot)
+        {
+            var dataTable = new DataTable();
+            var firstPass = true;
+            foreach (var element in dataRoot.EnumerateArray())
+            {
+                if (firstPass)
+                {
+                    foreach (var col in element.EnumerateObject())
+                    {
+                        var colValue = col.Value;
+                        dataTable.Columns.Add(new DataColumn(col.Name, colValue.ValueKind.ValueKindToType(colValue.ToString())));
+                    }
+                    firstPass = false;
+                }
+                var row = dataTable.NewRow();
+                foreach (var col in element.EnumerateObject())
+                {
+                    row[col.Name] = col.Value.JsonElementToTypedValue();
+                }
+                dataTable.Rows.Add(row);
+            }
+            return dataTable;
+        }
+
+        private static Type ValueKindToType(this JsonValueKind valueKind, string value)
+        {
+            switch (valueKind)
+            {
+                case JsonValueKind.String:      // 3
```

Comment:
What do these commented numbers mean?

Reply:
The integer value of JsonValueKind.  I'll remove the comments.

Reply:
Enum values, I'm removing the comments.